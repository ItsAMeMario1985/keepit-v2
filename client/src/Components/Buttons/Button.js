import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Button.propTypes = {
  onClick: PropTypes.func,
  buttonText: PropTypes.string,
}

export default function Button({ onClick, children }) {
  const bgColor = 'var(--color-primary)'

  return (
    <StyledButton bgColor={bgColor} onClick={onClick}>
      {children}
    </StyledButton>
  )
}

const StyledButton = styled.button`
  background-color: ${(props) => props.bgColor};
  border: none;
  min-height: 40px;
  min-width: 100px;
  width: 100%;
  font-weight: 400;
  color: white;
  padding: 8px 20px 8px 20px;
  border-radius: 5px;
  text-transform: uppercase;
  letter-spacing: 1px;

  svg {
    margin-left: 10px;
  }

  &:active {
    transform: scale(0.95);
    transition: transform 0.1s ease-out;
  }
`
