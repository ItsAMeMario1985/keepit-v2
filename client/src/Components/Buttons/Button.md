```js
<Button buttonText="Test" bgColor="orange" onClick={() => {}} icon="" />
```
