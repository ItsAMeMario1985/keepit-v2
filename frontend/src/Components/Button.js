import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

export default function Button({ onClick, buttonText }) {
  Button.propTypes = {
    onClick: PropTypes.func.isRequired,
    buttonText: PropTypes.string.isRequired,
  }
  return <button onClick={onClick}>{buttonText}</button>
}
