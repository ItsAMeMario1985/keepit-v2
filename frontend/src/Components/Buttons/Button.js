import styled from 'styled-components/macro'

export default function Button({ onClick, buttonText }) {
  const bgColor = 'var(--color-primary)'

  return (
    <StyledButton bgColor={bgColor} onClick={onClick}>
      {buttonText}
    </StyledButton>
  )
}

const StyledButton = styled.button`
  background-color: ${(props) => props.bgColor};
  font-weight: 400;
  color: white;
  min-width: 100px;
  padding: 8px 20px 8px 20px;
  border-radius: 5px;
  text-transform: uppercase;
  letter-spacing: 1px;

  svg {
    margin-left: 10px;
  }
`
