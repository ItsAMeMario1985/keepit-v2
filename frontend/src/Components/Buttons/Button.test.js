import { render } from '@testing-library/react'
import Button from './Button'
describe('Button', () => {
  it('renders correctly', () => {
    const { container } = render(<Button />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
