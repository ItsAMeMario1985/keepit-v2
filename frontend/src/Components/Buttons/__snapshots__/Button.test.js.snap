// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<span
  class="Button__StyledButton-sc-6pjoi6-0 dJFaRa"
/>
`;
